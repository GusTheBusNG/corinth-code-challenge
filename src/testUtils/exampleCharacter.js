export const character = {
  name: 'Anakin Skywalker',
  height: '6',
  mass: 'something',
  hair_color: 'brown',
  birth_year: 'some time long ago',
  species: ['example url species'],
  starships: ['example url starships'],
  films: ['example url fil1s']
}
