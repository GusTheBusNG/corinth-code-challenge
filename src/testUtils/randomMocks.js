export const stubLoading = (message = 'loading') => {
  const random = jest.requireActual('../utils/randomPhrases');
  random.getLoadingQuote = jest.fn(() => message)
}

export const stubError = (message = 'error') => {
  const random = jest.requireActual('../utils/randomPhrases');
  random.getErrorQuote = jest.fn(() => message)
}