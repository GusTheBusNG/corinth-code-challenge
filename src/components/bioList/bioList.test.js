import React from 'react';
import { create } from 'react-test-renderer';

import BioList from './';
import { stubLoading } from '../../testUtils/randomMocks'

const DEFAULT_PROPS = {
  label: 'label',
  requestUrls: ['urls']
}

const createBioList = customProps => 
  create(<BioList {...{...DEFAULT_PROPS, ...customProps}} />);


describe('BioList', () => {
  describe('render', () => {
    it('renders correctly', () => {
      stubLoading()

      expect(createBioList().toJSON()).toMatchSnapshot();
    })
  })
})
