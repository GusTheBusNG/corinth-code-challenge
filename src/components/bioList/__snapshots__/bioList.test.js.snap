// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BioList render renders correctly 1`] = `
<div
  className="info-pill"
>
  <img
    alt="label"
    className="icon"
    src="https://res.cloudinary.com/dnglix0tu/image/upload/f_auto/v1/LifeWay-star-wars/label?_a=ATAMhAA0"
  />
  <p
    className="content"
  >
    loading
  </p>
</div>
`;
